ledger-service/lf-value-json/src/main/scala/com/digitalasset/daml/lf/value/json/ApiCodecCompressed.scala

```diff
@@ -136,7 +136,13 @@ class ApiCodecCompressed[Cid](val encodeDecimalAsString: Boolean, val encodeInt6
       case Model.DamlLfPrimType.Timestamp => { case JsString(v) =>
         V.ValueTimestamp(assertDE(Time.Timestamp fromString v))
       }
-      case Model.DamlLfPrimType.Date => { case JsString(v) => V.ValueDate.fromIso8601(v) }
+      case Model.DamlLfPrimType.Date => { case JsString(v) =>
+        try {
+          V.ValueDate.fromIso8601(v)
+        } catch {
+          case _: Exception => throw DeserializationException(s"Invalid date: $v")
```

Comment:
The table of failures in `ApiCodeCompressedSpec` expects `DeserializationException` specifically. I wasn't sure which side should be changed.

Reply:
Changing here seems fine but I’d prefer to catch `java.time.format.DateTimeParseException` and `IllegalArgumentException` instead of catching `Exception`.

Reply:
I slightly disagree but since you're not here to argue today I'll go with your approach.